src/containers/ProductPage/ProductPage.js

```diff
@@ -0,0 +1,122 @@
+import Button from "../Button/Button";
+import "./ProductPage.css";
+import { NavLink } from "react-router-dom";
+import { useParams } from "react-router-dom";
+import { useContext } from "react";
+import CartContext from "../../context/CartContext";
+
+
+function ProductPage(products) {
+  const {CartProducts,handleProducts} = useContext(CartContext)
+  let params = useParams();
+  let idItem = params.id;
+  let results = products.Products.data.results;
+  let product = getProduct(results, idItem);
+  if (product === null)
+    return (
+      <div>
+        <h1>Product not Found.</h1>
+      </div>
+    );
+  let title = product.data.name;
+  let img = product.data.mainimage.url;
+  let desc = product.data.short_description;
+  let price = product.data.price;
+  let category = product.data.category.slug;
+  let sku = product.data.sku;
+  let tags = product.tags;
+  let specs = product.data.specs;
+  let id = product.id;
+  let stock = product.data.stock;
+  
+  const getQtyInCart=()=>{
+    let cont=0;
+    CartProducts.forEach((element,i) => {
+      if(id===element.id){
+      cont=element.qty;
+      }
+  });
+  return cont;
+  }
+
+  return (
+    <div style={{ display: "flex", flexDirection: "row" }}>
+      <div class="productPageCard">
+        <div class="productPagetitle">
+          <h1 class="productPagetitle">
+            [{category}] {title}
+          </h1>
+          <h1> $ {price}</h1>
+        </div>
+
+        <div style={{ display: "flex", flexDirection: "row" }}>
+          <div>
+            <img class="productPageitemimg" src={img} />
+          </div>
+          <div style={{ display: "flex", flexDirection: "column" }}>
+            <div><h3>Description: <p>{desc}</p></h3></div>
+            <h3>SKU: {sku}</h3>
+
+            <div>
+              <h2>
+                Qty <input type="text" value={getQtyInCart()}  ></input>
+                {getQtyInCart()<stock?<button onClick={()=>{handleProducts({id:id,title:title,stock:stock,category:category,price:price,desc:desc,qty:1},1)}}>+1</button>:""}
```

Comment:
* I think for this layout is better to disable the button instead of hiding it, since you can press accidentally the other button
* lets define the object upside in the code instead of writing twice the object literal
 ```javascript 
let productToHandle = {id, title, stock, category, price,desc, qty:1} 
```
then 
```javascript
onClick={() => {handleProducts(productToHandle,1);}}
```